source/h2_stream.c

```diff
@@ -812,6 +828,11 @@ int aws_h2_stream_encode_data_frame(
             stream->thread_data.state = AWS_H2_STREAM_STATE_HALF_CLOSED_LOCAL;
             AWS_H2_STREAM_LOG(TRACE, stream, "Sent END_STREAM. State -> HALF_CLOSED_LOCAL");
         }
+
+        AWS_ASSERT(stream->base.metrics.send_end_timestamp_ns == -1);
+        aws_high_res_clock_get_ticks((uint64_t *)&stream->base.metrics.send_end_timestamp_ns);
+        stream->base.metrics.sending_duration_ns =
+            stream->base.metrics.send_end_timestamp_ns - stream->base.metrics.send_start_timestamp_ns;
```

Comment:
Should we move this to start of the if block like before so that it is set in the error case as well?